inference/models/inference.py

```diff
@@ -0,0 +1,292 @@
+                status=ExternalJobStatus.QUEUED,
+            )
+
+        except Exception as e:
+            self.set_error(f"Failed to create external job: {str(e)}")
+            return None
+
+    def process_external_jobs(self) -> None:
+        """Process all pending external jobs"""
+        pending_jobs = self.external_jobs.filter(status__in=[ExternalJobStatus.QUEUED, ExternalJobStatus.PENDING])
+
+        for external_job in pending_jobs:
+            external_job.process()
+
+    def evaluate_status(self) -> None:
+        """Evaluate overall job status and handle completion"""
+
+        if self.is_ongoing:
+            self.status = InferenceJobStatus.PENDING
+        else:
+            failed_jobs = self.get_failed_jobs()
+            if failed_jobs.exists():
+                self.status = InferenceJobStatus.FAILED
+            else:
+                self.status = InferenceJobStatus.COMPLETED
+                self.cleanup()
+
+            self.completed_at = timezone.now()
+
+        self.save()
+
+    def cleanup(self) -> None:
+        """Handle cleanup after job completion"""
+        try:
+            # Unload model if no other jobs need it
+            if not InferenceJob.objects.filter(
+                model_version=self.model_version, status=InferenceJobStatus.PENDING
+            ).exists():
+                model_manager = ModelManager(InferenceAPIClient(), self.model_version.api_identifier)
+                model_manager.api_client.unload_model(self.model_version.api_identifier)
+
+        except Exception as e:
+            self.set_error(f"Cleanup error: {str(e)}")
+
+
+class ExternalJob(models.Model):
+    """
+    Represents a batch job sent to the inference API.
+    Multiple ExternalJobs can belong to one InferenceJob.
+    """
+
+    inference_job = models.ForeignKey(InferenceJob, on_delete=models.CASCADE, related_name="external_jobs")
+    external_job_id = models.CharField(max_length=255, help_text="Job ID returned by the inference API")
+
+    url_ids = models.JSONField(help_text="List of URL IDs included in this batch")
+
+    status = models.IntegerField(choices=ExternalJobStatus.choices, default=ExternalJobStatus.QUEUED)
+    results = models.JSONField(blank=True)
+    error_message = models.TextField(blank=True)
+
+    created_at = models.DateTimeField(default=timezone.now)
+    updated_at = models.DateTimeField(auto_now=True)
+    completed_at = models.DateTimeField(null=True, blank=True)
+
+    def set_status(self, status: ExternalJobStatus) -> None:
+        """Update job status"""
+        self.status = status
+        self.save(update_fields=["status", "updated_at"])
+
+    @property
+    def is_ongoing(self) -> bool:
+        """Check if job is currently processing"""
+        return self.status in [ExternalJobStatus.QUEUED, ExternalJobStatus.PENDING]
+
+    def mark_completed(self):
+        """Mark batch as completed and check parent job completion"""
```

Comment:
Leave note that in order for an ExternalJob to exist, it must have already been created in the InferenceApi